fineract-provider/src/main/java/org/apache/fineract/infrastructure/crypt/utils/RSAEncryptionUtils.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.crypt.utils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Random;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.xml.bind.DatatypeConverter;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.fineract.infrastructure.core.exception.GeneralPlatformDomainRuleException;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.infrastructure.crypt.domain.EncryptionKeys;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author manoj
+ */
+@Component
+public class RSAEncryptionUtils {
+    private final static Logger LOG = LoggerFactory.getLogger(RSAEncryptionUtils.class);
+
+    public EncryptionKeys generateKeys(){
+        KeyPairGenerator keyPairGenerator;
+        try {
+            keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+
+            KeyPair keyPair = keyPairGenerator.generateKeyPair();
+            PublicKey publicKey = keyPair.getPublic();
+            PrivateKey privateKey = keyPair.getPrivate();
+            byte[] publicKeyAsBytes = publicKey.getEncoded();
+            byte[] privateKeyAsBytes = privateKey.getEncoded();
+
+            return new EncryptionKeys(privateKeyAsBytes, publicKeyAsBytes, DateUtils.getLocalDateTimeOfTenant(), generateVersionString());
+
+        } catch (NoSuchAlgorithmException e) {
+           LOG.error("Error occurred", e);
+        }
+        throw new GeneralPlatformDomainRuleException("error.msg.error.in.generating.keys", "Error in generating keys");
```

Comment:
You could avoid this by doing the `keyPairGenerator = KeyPairGenerator.getInstance("RSA");` in the constructor, and let this error propagate there? This would cause it to fail at boot, which would be better. If you do want to keep it, then IMHO it would be better like this, even though it does the exact same thing, just because clearer to understand:

```suggestion
        } catch (NoSuchAlgorithmException e) {
           LOG.error("Error occurred", e);
           throw new GeneralPlatformDomainRuleException("error.msg.error.in.generating.keys", "Error in generating keys");
        }
```

Reply:
Changed this.  I was just being sure that the method return something or an exception(do or die).Yet you are right, there is better ways to do it rather than sending panic